client/src/featureform/register.py

```diff
@@ -1995,14 +2054,14 @@ def get_entity(self, name, local=False):
     def register_redis(
         self,
         name: str,
+        host: str,
+        port: int,
+        password: str,
+        db: int,
         description: str = "",
         team: str = "",
-        host: str = "0.0.0.0",
-        port: int = 6379,
-        password: str = "",
-        db: int = 0,
-        tags: List[str] = [],
-        properties: dict = {},
+        tags: List[str] = None,
```

Comment:
ideally these are all `field(default_factory=list)` or `field(default_factory=dict)` that way we never deal with them being none